api/v1beta1/keystoneapi_types.go

```diff
@@ -128,19 +128,19 @@ type KeystoneAPISpec struct {
 
 	// +kubebuilder:validation:Optional
 	// NetworkAttachments is a list of NetworkAttachment resource names to expose the services to the given network
-	NetworkAttachments []string `json:"networkAttachments"`
+	NetworkAttachments []string `json:"networkAttachments,omitempty"`
```

Comment:
I'm wondering, would a `// +kubebuilder:default=[]` solve this too? If so then I would go that way as that would actually document to the API user what will be the value of the field if it is not provided.

Reply:
works for me if that is ok for the work @rabi is doing right now.